src/events.rs

```diff
@@ -1,22 +1,71 @@
 use crate::context::AudioNodeId;
+use crossbeam_channel::Receiver;
+use std::sync::{Arc, Mutex};
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
-pub(crate) enum EventType {
-    Ended,
+pub(crate) enum Event {
+    Ended(AudioNodeId),
+    SinkChanged,
+}
+
+pub(crate) enum Callback {
+    Once(Box<dyn FnOnce() + Send + 'static>),
+    Multiple(Box<dyn FnMut() + Send + 'static>),
+}
+
+impl Callback {
+    fn run(self) {
+        match self {
+            Self::Once(f) => (f)(),
+            Self::Multiple(mut f) => (f)(),
+        }
+    }
 }
 
 pub(crate) struct EventHandler {
-    // could be optional meaning that its a context event (cf. onSinkChange, onStateChange, etc.)
-    pub node_id: AudioNodeId,
-    pub event_type: EventType,
-    pub callback: Box<dyn Fn() + Send + Sync + 'static>,
+    pub event: Event,
+    pub callback: Callback,
+}
```

Comment:
Not sure the naming is really consistent with the spec, as far as I understand it, in the spec `EventHandler` is what is called `Callback` here. Probably not really important as it is `pub(crate)` but could be confusing

Reply:
Valid point, I will rename them

Reply:
I just realized we already use the name EventHandler:

```
pub(crate) struct EventHandler {
    pub event: Event,
    pub callback: Callback,
}
```

It is the callback combined with the type of event we are expecting for the callback. So I'm leaving it this way.